object-file.c

```diff
@@ -1773,14 +1773,14 @@ void *read_object_with_reference(struct repository *r,
 }
 
 static void write_object_file_prepare(const struct git_hash_algo *algo,
-				      const void *buf, unsigned long len,
+				      const void *buf, size_t len,
 				      const char *type, struct object_id *oid,
-				      char *hdr, int *hdrlen)
+				      char *hdr, size_t *hdrlen)
 {
 	git_hash_ctx c;
 
 	/* Generate the header */
-	*hdrlen = xsnprintf(hdr, *hdrlen, "%s %"PRIuMAX , type, (uintmax_t)len)+1;
+	*hdrlen = xsnprintf(hdr, *hdrlen, "%s %"PRIuMAX , type, (size_t)len)+1;
```

Comment:
This change is incorrect. `PRIuMAX` _does_ take a `uintmax_t`, whatever that might be. For example, on 32-bit systems a `size_t` will be 32-bit but an `unsigned long long` might very well _still_ be 64-bit.

This is what https://github.com/git-for-windows/git/runs/4199379908?check_suite_focus=true#step:5:439 tried to point out.

Reply:
Thanks. That change was mid session of trying to determine what was happening. I'd done the higher level changes, but was still getting sha1 errors, so somewhat guessed at that one. 

story: In my mind's eye I had that the length value (in the loose blob object) was a fixed format - probably seeing too many discussion of the network packet format. I now see that the length is null terminated, rather than fixed format. I was also minded that I was expecting, on Windows that `size_t` converts to `unsigned long long`, and that would be the same as the `PRIuMAX` format. How wrong I was. print format strings can be tricky (tricked me)...

When I got past the other changes I needed to get the sha1 to evaluate correctly, and then pushed to GitHub, I also had mistakenly the docker failure as being the same as the other CI failure that had been mentioned on list, so didn't investigate properly, as it was all working elsewhere.. 
/story.

Reply:
.. Yes, to confirm, I'll revert that change.